pkg/app/piped/eventwatcher/eventwatcher.go

```diff
@@ -388,7 +391,7 @@ func (w *watcher) execute(ctx context.Context, repo git.Repo, repoID string, eve
 			switch handler.Type {
 			case config.EventWatcherHandlerTypeGitUpdate:
 				branchName, err := w.commitFiles(ctx, latestEvent, matcher.Name, handler.Config.CommitMessage, e.GitPath, handler.Config.Replacements, tmpRepo, handler.Config.MakePullRequest)
-				if err != nil {
+				if err != nil && !errors.Is(err, errNoChanges) {
```

Comment:
[IMO] How about using the bool variable like `change` to minimize the scope of the error variable and also for the readability?

```suggestion
				noChange := false
				if err != nil {
				    // 
				    if errors.Is(err, errNoChanges) {
				        noChange = true
				    } else {
				     ...   
				     continue
				    }
				}
				
				if noChange {
				        handledEvent.StatusDescription = "Nothing to commit"
					gitNoChangeEvents = append(gitNoChangeEvents, handledEvent)
				} else {
```

Reply:
thank you , got it, i fixed in [7b155e6](https://github.com/pipe-cd/pipecd/pull/5310/commits/7b155e62a7e7a994b97e6b4ac31a6d082cd415f5)